src/main/java/org/jabref/logic/util/BasicFileType.java

```diff
@@ -0,0 +1,161 @@
+    PLAIN_XMP(Localization.lang("%0 file", "XMP"), "xmp"),
+
+    AUX(Localization.lang("%0 file", "AUX"), "aux"),
+    JSTYLE(Localization.lang("Style file"), "jstyle"),
+    LAYOUT(Localization.lang("Custom layout file"), "layout"),
+    TERMS(Localization.lang("Protected terms file"), "terms"),
+    TXT(Localization.lang("%0 file", Localization.lang("Plain text")), "txt"),
+    CLASS(Localization.lang("%0 file", "CLASS"), "class"),
+    JAR(Localization.lang("%0 file", "JAR"), "jar"),
+    XML(Localization.lang("%0 file", "XML"), "xml"),
+    ZIP(Localization.lang("%0 file", "ZIP"), "zip"),
+    ODS(Localization.lang("%0 file", Localization.lang("OpenDocument spreadsheet")), "ods"),
+    SXC(Localization.lang("%0 file", "OpenOffice/LibreOffice Calc"), "sxc"),
+    HTML(Localization.lang("%0 file", "HTML"), "html"),
+    RTF(Localization.lang("%0 file", "RTF"), "rtf"),
+    RDF(Localization.lang("%0 file", "RDF"), "rdf"),
+    CSV(Localization.lang("%0 file", "CSV"), "csv"),
+    DEFAULT(Localization.lang("%0 file", "DEFAULT"), "default");
+
+    private static Set<FileType> allfileTypes = new HashSet<>(EnumSet.allOf(BasicFileType.class));
+    private final String[] extensions;
+    //Needed for custom exporters
+    private String description;
+
+    BasicFileType(String description, String... extensions) {
+        this.description = description;
+        this.extensions = extensions;
+    }
+
+    @Override
+    public List<String> getExtensions() {
+        return getExtensions(extensions);
+    }
+
+    @Override
+    public String getDescription() {
+        return getDescription(description, extensions);
+    }
+
+    @Override
+    public String getFirstExtensionWithDot() {
+        return getFirstExtensionWithDot(extensions);
+    }
+
+    @Override
+    public List<String> getExtensionsWithDot() {
+        return getExtensionsWithDot(extensions);
+    }
+
+    public static FileType addnewFileType(String description, String... extensions) {
+        FileType fileType = new FileType() {
+
+            @Override
+            public String getFirstExtensionWithDot() {
+                return BasicFileType.getFirstExtensionWithDot(extensions);
+            }
+
+            @Override
+            public List<String> getExtensionsWithDot() {
+                return BasicFileType.getExtensionsWithDot(extensions);
+            }
+
+            @Override
+            public List<String> getExtensions() {
+                return BasicFileType.getExtensions(extensions);
+            }
+
+            @Override
+            public String getDescription() {
+                return BasicFileType.getDescription(description, extensions);
+            }
+        };
+        allfileTypes.add(fileType);
+        return fileType;
+
+    }
+
+    public static void addnewFileType(FileType fileType) {
+        allfileTypes.add(fileType);
+    }
+
+    public static FileType parse(String description) {
```

Comment:
It appears to me that the `parse` method is no longer needed and I suspect that hence the `allFileTypes` list can be removed. Right?

Reply:
Yes missed that.